objects.go

```diff
@@ -128,6 +170,35 @@ func (chain *Chain) writeOperation(verb verb, ctx *nftContext, writer io.Writer)
 	fmt.Fprintf(writer, "\n")
 }
 
+// groups----------------------------1-2--3------4-------5-6---------7----8----------9-----10--------11---------
+var chainExp = fmt.Sprintf(` %s( {( type %s hook %s( device "%s")?( priority %s ;))?( comment %s ;)? })?`,
+	noSpaceGroup, noSpaceGroup, noSpaceGroup, noSpaceGroup, noSpaceGroup, commentGroup)
```

Comment:
The numbers aren't aligned correctly are they? Like, shouldn't `1` be over the first `%s`?

Also, if you're only using `()` because you need to put a `?` after it, and you don't actually care about capturing the contents of that group in the `match` result, you can use `(?:XXX)` rather than `(XXX)` to create a "non-capturing group".

Reply:
different fonts don't work well, updated to one-line numbering